Tests/AppStoreConnectClientTests/RateLimitMiddlewareTests.swift

```diff
@@ -0,0 +1,168 @@
+// FS App Store Connect Client
+// Copyright (C) 2024  FREEDOM SPACE, LLC
+
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Affero General Public License as published
+//  by the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU Affero General Public License for more details.
+//
+//  You should have received a copy of the GNU Affero General Public License
+//  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+//
+//  RateLimitMiddlewareTests 2.swift
+//
+//
+//  Created by Mykola Vasyk on 15.05.2024.
+//
+
+import XCTest
+import HTTPTypes
+import OpenAPIRuntime
+@testable import AppStoreConnectClient
+
+class RateLimitMiddlewareTests: XCTestCase {
+    func testExtractHeaderValueSuccess() throws {
+        let sut = RateLimitMiddleware()
+        var fields = HTTPFields()
+        let unwrap = try XCTUnwrap(HTTPField.Name("x-rate-limit"))
+        fields[unwrap] = "user-hour-lim:1111; user-hour-rem:222"
+        let mockResponse = HTTPTypes.HTTPResponse(
+            status: .accepted,
+            headerFields: HTTPFields(fields[fields: unwrap])
+        )
+        XCTAssertNoThrow(try sut.extractHeaderValue(from: mockResponse, for: "x-rate-limit"))
```

Comment:
extractHeaderValue must be private and therefore this test is not correct

Reply:
Changed.